ignite/handlers/checkpoint.py

```diff
@@ -224,7 +229,17 @@ def __call__(self, engine: Engine) -> None:
     def _setup_checkpoint(self) -> dict:
         checkpoint = {}
         for k, obj in self.to_save.items():
-            checkpoint[k] = obj.state_dict()
+            if isinstance(obj, torch.nn.parallel.DistributedDataParallel):
+                # only save in first process
+                if torch.distributed.get_rank() > 0:
```

Comment:
This point is very interesting to handle. Previously, in the examples, we show that usage of `Checkpoint` is conditioned by `torch.distributed.get_rank() > 0`. But we can say that all our classes can handle that internally. Similar to what is done here. But maybe it is need in `__call__` method...
So, for instance, I think we do not need this here.